src/gui/shell.cpp

```diff
@@ -867,6 +866,38 @@ void Shell::handleGuiFontFunction(const QVariantList& args)
 	setGuiFont(fdesc, force, true /*setOption*/);
 }
 
+void Shell::handleCloseEvent(const QVariantList &args)
+{
+	qDebug() << "Neovim requested a GUI close";
+
+	// Decide which exit status Neovim-qt should exit with.
+	int status{};
+	if (args.size() == 1) {
```

Comment:
You could remove this block entirely.

Just initialize status to 0: `int status{ 0 };`